src/rvsdg/to_cfg.rs

```diff
@@ -154,84 +131,157 @@ impl RvsdgFunction {
             }
         }
 
-        to_bril.operand_to_bril(self.state, &rvsdg_args, &None);
+        let mut result = to_bril.operand_to_bril(
+            self.state,
+            &rvsdg_args,
+            &RvsdgContext {
+                body: None,
+                branch: 0,
+            },
+        );
         if let Some((_ty, operand)) = &self.result {
             // it doesn't matter what var we assign to
             // TODO current args hardcoded to implicit print state
-            let res = to_bril.operand_to_bril(*operand, &rvsdg_args, &None);
-            to_bril.current_instrs.push(Instruction::Effect {
+            let return_res = to_bril.operand_to_bril(
+                *operand,
+                &rvsdg_args,
+                &RvsdgContext {
+                    body: None,
+                    branch: 0,
+                },
+            );
+            let final_block = to_bril.make_block(vec![Instruction::Effect {
                 op: EffectOps::Return,
-                args: vec![res.unwrap_name()],
+                args: vec![return_res.get_single_res().unwrap_name()],
                 funcs: vec![],
                 labels: vec![],
                 pos: None,
-            });
+            }]);
+
+            result = to_bril.sequence_results(&[
+                result,
+                return_res,
+                TranslationResult {
+                    start: final_block,
+                    end: final_block,
+                    values: vec![],
+                },
+            ]);
         }
-        let last_block = to_bril.finish_block();
 
         // TODO hard-coded name
         CfgFunction {
             name: "main".into(),
             args: func_args,
             graph: to_bril.graph,
-            entry: to_bril.entry_block.unwrap(),
-            exit: last_block,
+            entry: result.start,
+            exit: result.end,
             return_ty: None,
             _phantom: Simple,
         }
     }
 }
 
 impl<'a> RvsdgToCfg<'a> {
-    // Translates an operand to bril, returning the name
-    // of the variable (or a placeholder for state edges, which don't show up in bril).
+    /// Given two [`TranslationResult`]s, sequences them in the CFG
+    /// and returns a new [`TranslationResult`] with the same values as the second.
+    fn sequence_results(&mut self, results: &[TranslationResult]) -> TranslationResult {
+        for (res1, res2) in results.iter().zip(results.iter().skip(1)) {
+            self.graph.add_edge(
+                res1.end,
+                res2.start,
+                Branch {
+                    op: BranchOp::Jmp,
+                    pos: None,
+                },
+            );
+        }
+        TranslationResult {
+            start: results[0].start,
+            end: results[results.len() - 1].end,
```

Comment:
nit, optional: I think doing `results.last().unwrap()` is a bit easier to read than `results[results.len()-1]`.

Reply:
done